aws/validators.go

```diff
@@ -2424,3 +2424,31 @@ func validateRoute53ResolverName(v interface{}, k string) (ws []string, errors [
 
 	return
 }
+
+func validateGlueCatalogDatabaseName(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if regexp.MustCompile(`[A-Z]`).MatchString(value) {
+		errors = append(errors, fmt.Errorf("%q can only contain lowercase alphanumeric characters", k))
+	}
+	if len(value) < 1 {
+		errors = append(errors, fmt.Errorf("%q must be at least 1 character long", k))
+	}
+	if len(value) > 255 {
+		errors = append(errors, fmt.Errorf("%q cannot be more than 255 characters long", k))
+	}
+	return
```

Comment:
Maybe also validate the whole string against a validity regex to stop special characters getting through. e.g.
```
^[a-z0-9_-]*$
```

Reply:
or even better use AWS's 
https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-catalog-databases.html
https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-common.html#aws-glue-api-regex-oneLine
```
[\u0020-\uD7FF\uE000-\uFFFD\uD800\uDC00-\uDBFF\uDFFF\t]*
```

Reply:
Should that regex be valid in golang? I don't have much experience with using unicode characters in regular expressions but here's the error I'm getting. https://play.golang.org/p/HB2sOLvpRoF
```
panic: regexp: Compile(`(?m)[\u0020-\uD7FF\uE000-\uFFFD\uD800\uDC00-\uDBFF\uDFFF\t]*`): error parsing regexp: invalid escape sequence: `\u`

```

Reply:
Additionally, special characters are in fact allowed in both the table and database names. I do however need to account for other cases provided by the regex you pointed me to so if we can fix that this PR should be ready.

Reply:
I don't think that regex is valid, `\uDC00-\uDBFF` is an invalid range. I only have minimal experience using unicode expressions but as long as I've converted to go notation correctly I think we've found a bug.
https://play.golang.org/p/jhV8pEEjPrT

Reply:
Additionally, special characters are in fact allowed in both the table and database names. Technically, Uppercase is allowed in the api call as well but it's just converted to lowercase after.
The reason for only accepting lowercase in terraform is because otherwise terraform would always attempt to reapply changes. I think my validator works but if you'd like me to make changes I can.

Reply:
No, I think your right :+1: